### PR TITLE
Bump basalt to v0.2.1

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-h5py/package.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class PyH5py(PythonPackage):
 
     import_modules = ['h5py', 'h5py._hl']
 
+    version('2.9.0', sha256='9d41ca62daf36d6b6515ab8765e4c8c4388ee18e2a665701fef2b41563821002')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-h5py/package.py:18]: Any particular reason for this?

File: var/spack/repos/builtin/packages/py-numpy/package.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ class PyNumpy(PythonPackage):
         'numpy.distutils.command', 'numpy.distutils.fcompiler'
     ]
 
+    version('1.16.2', sha256='6c692e3879dde0b67a9dc78f9bfb6f61c666b4562fd8619632d7043fb5b691b0')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-numpy/package.py]: And this? The two changes rebuild a lot of stuff. Is there a feature we cannot live without?

File: var/spack/repos/builtin/packages/py-h5py/package.py (hunk 1 above)
Comment (tristan0x) [reply var/spack/repos/builtin/packages/py-h5py/package.py:18]: Yes, I need support for float128. Tests fail with 2.8.0

File: var/spack/repos/builtin/packages/py-numpy/package.py (hunk 2 above)
Comment (tristan0x) [reply var/spack/repos/builtin/packages/py-numpy/package.py]: Actually, tests pass with the current version. I will remove this change from this PR.

File: var/spack/repos/builtin/packages/py-h5py/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-h5py/package.py:18]: OK

**Review by matz-e — APPROVED** [review]
LGTM

**Pull request merged**